index.js

```diff
@@ -27,10 +32,32 @@ const buildPath = path.join(__dirname, "stocks-frontend", "build");
 
 app.use(express.static(buildPath));
 
-app.get("/", (request, response) => {
-  response.sendFile(path.join(buildPath, "index.html"));
+app.get("/", (req, res) => {
+	res.sendFile(path.join(buildPath, "index.html"));
+});
+
+app.get("/stocks", (req, res) => {
+	// each request has a parameter called query that holds eveything from ? in the url
+	// query must have one param - symbol
+	if (req.query && req.query != {}) {
```

Comment:
Express guarantees `req.query` to be a valid object, so `req.query` should always be truthy. 

`req.query != {}` will always evaluate to `true` since it's just comparing the addresses of the objects instead of their properties (the RHS is an anonymous object, so they will always have different addresses).

Instead of attempting to ensure that the `req.query` object is valid and has a nonzero number of properties, we should just directly check the type of `req.query.symbol` (if it's not defined, this will evaluate to `false`). 


```js
if (typeof req.query.symbol === 'string') {
    const symbol = req.query.symbol;
    const results = await stocksApi.getStockData(symbol);
    req.send(results);
} else {
    res.status(400).send("please add query param for symbol");
}
```
This has the benefit of removing an extra branch, and it handles the case where `symbol` is defined multiple times in the query string (this would evaluate to an array instead of a string, which would then cause `getStockData` to choke).

Reply:
what if the symbol is not sent? `req.query.symbol` will fail. I will add a `"symbol" in query && typeof req.query.symbol === 'String'` if that is fine?

Reply:
Cool!

Reply:
If "symbol" is not in the URL query parameters, then `req.query.symbol` will have the value of  `undefined`. `typeof undefined` gives you the string `"undefined"`, which will not compare equal with the string `"string"`.

So, there shouldn't be a need to check if `query` actually contains the property `symbol` with the `in` keyword.

Also, make sure the compared string is a lowercase `"string"`; if it's uppercase, then the comparison will always fail.

Reply:
Makes sense. Made the changes